Implement serialization

sallesma opened the pull request:
### Summary

Implement serialization from ruby array structure to EDIFACT.

### Other Information

I gave it a try after creating issue #1
Not sure it is what you intend to build with this gem.
If needed I am open to suggestions to improve the code in this PR

sallesma commented:
Hi @david-blackwood 
Do you think this PR has any chances to be merged or reviewed?

david-blackwood commented on lib/edifact_rails/serializer.rb:
```diff
@@ -0,0 +1,57 @@
+
+module EdifactRails
+  class Serializer
+    # Treat the input a little, split the input string into segments, parse them
```
remove copied comment

david-blackwood commented on spec/edifact_rails_spec.rb:
```diff
@@ -93,34 +93,140 @@
 
     expect(result).to eq(expected)
   end
+
```
Add a test for the `with_service` option

david-blackwood commented on lib/edifact_rails/serializer.rb:
```diff
@@ -0,0 +1,57 @@
+  class Serializer
+    # Treat the input a little, split the input string into segments, parse them
+    def serialize(segments, with_service:)
+      # Add the UNA segment
```
move this UNA comment down to line 13, perhaps quickly explain purpose of `with_service` parameter here

david-blackwood commented:
Hi Sallasma,

Thank you for your work: it's great stuff.

I have requested a couple of very minor changes before I merge this in.

Thanks,
David

sallesma commented:
Thanks for the review.
I've made some changes to make it better according to your suggestions

david-blackwood commented:
Hello again Sallesma:

I have added UNA support to the parser. This has some minor ramifications on your pr. Please update your pr and ensure it is merge-compatible, and I will be happy to merge your work in.

Apologies for the delay.

sallesma commented:
Hi David,

Sorrry I haven't had much time in the past weeks. I have rebased my branched against master locally, and fixed the conflicts, but I see that you have changed the way special characters are handled, with custom code inside the Parser class instead of constants. I don't feel like it should be duplicated to the Seriaalizer class, do you have any idea how you would like it done?

sallesma commented:
In the end I just used the defaults special characters

david-blackwood commented:
This work has been merged and pushed with v2.1.0.

There is some consideration to be made for accepting custom special characters as a hash, or even specifying other formats and outputting as them, but that is not needed for now.

Thank you very much for your contribution!
